src/realm/sync/instruction_replication.cpp

```diff
@@ -564,13 +559,18 @@ void SyncReplication::list_move(const CollectionBase& view, size_t from_ndx, siz
     }
 }
 
-void SyncReplication::list_erase(const CollectionBase& view, size_t ndx)
+void SyncReplication::list_erase(const CollectionBase& list, size_t ndx)
 {
-    size_t prior_size = view.size();
-    Replication::list_erase(view, ndx);
-    if (select_collection(view)) {
+    Mixed prior_value = list.get_any(ndx);
```

Comment:
Was it possible for us to previously emit an ArrayErase instruction for an unresolved link? I tried to test this but couldn't make it happen.

Reply:
If you called `obj.get_list<ObjKey>(col_list).remove(5)` and the fifth (unfiltered) element was unresolved then you would get the instruction in the log. But it should not be possible to make it happen using the public user API.

Reply:
I thought deleting an object which has a list with unresolved links might trigger this, but it doesn't seem to be the case.

Reply:
@jedelbo sorry I missed this earlier, but shouldn't we have the same checks in `SyncReplication::link_list_nullify` and `SyncReplication::nullify_link` ?